version: '3'
services:
  db:
    image: mysql:5.7
    ports:
      - "3306:3306" # <-これがないと外からアクセス出来ない
    container_name: pubz-db-dev
    hostname: db
    environment:
      MYSQL_ROOT_PASSWORD: $MYSQL_ROOT_PASSWORD
      MYSQL_DATABASE: $MYSQL_DATABASE
      MYSQL_USER:     $MYSQL_USER
      MYSQL_PASSWORD: $MYSQL_PASSWORD
    volumes:
      - dev-storage:/var/lib/mysql
      - ./docker/mysql:/etc/mysql/conf.d
  phpmyadmin:
    image: phpmyadmin/phpmyadmin
    environment:
      PMA_ARBITRARY: 1
      PMA_HOST:      db
      PMA_USER:      root
      PMA_PASSWORD:  docker
    links:
      - db
    ports:
      - "7070:80"
  django:
    build: ./docker/django
    # command: python /code/app/manage.py runserver 0.0.0.0:8000
    container_name: pubz-web-dev
    hostname: pubz-web-dev
    # hostname: django
    environment:
      MYSQL_DATABASE: $MYSQL_ROOT_PASSWORD
      MYSQL_USER:     $MYSQL_USER
      MYSQL_PASSWORD: $MYSQL_PASSWORD
      MYSQL_HOST:     db
      # VIRTUAL_HOST: 'app.www-mmde.ist.osaka-u.ac.jp'
    volumes:
      - ./src:/code
      - ./docs:/root/docs
      - ./utils:/root/utils
    ports:
      - "7000:8000"
      - "7777:7777"
    depends_on:
      - db
    tty: true

volumes:
  dev-storage:
    driver: local
    driver_opts:
      type: none
      device: $PWD/storage/db/data/
      o: bind


networks:
  default:
    external:
      name: shared
